test/authorization/service/role.service.test.ts

```diff
@@ -78,170 +95,226 @@ describe('test Role Service', () => {
-
-    permissionQueryBuilder.getMany().resolves(permissions);
-
-    connectionMock.transaction(Arg.any()).resolves(request);
-
-    const resp = await roleService.updateRolePermissions(
-      'ae032b1b-cc3c-4e44-9197-276ca877a7f8',
-      input,
-    );
-    expect(resp).toEqual(permissions);
+  describe('updateRole', () => {
+    it('should update a role for a valid role id', async () => {
+      const input: UpdateRoleInput = {
+        name: 'Test1',
+      };
+
+      updateMock.mockResolvedValue(updateResult);
+
+      const result = await roleService.updateRole(
+        'ae032b1b-cc3c-4e44-9197-276ca877a7f8',
+        input,
+      );
+
+      expect(updateMock).toBeCalledWith(VALID_ROLE_ID, input);
+
+      expect(result).toEqual(true);
+    });
   });
 
-  it('should throw exception when adding invalid permissions to a role', async () => {
-    const input: UpdateRolePermissionInput = {
-      permissions: ['3e9e78c9-3fcd-4eed-b027-62f794680b03'],
-    };
-
-    permissionRepository
-      .findByIds(['3e9e78c9-3fcd-4eed-b027-62f794680b03'])
-      .resolves([]);
-
-    const resp = roleService.updateRolePermissions(
-      'ae032b1b-cc3c-4e44-9197-276ca877a7f8',
-      input,
-    );
-    await expect(resp).rejects.toThrowError(
-      new PermissionNotFoundException(
-        ['3e9e78c9-3fcd-4eed-b027-62f794680b03'].toString(),
-      ),
-    );
+  describe('deleteRole', () => {
+    it('should delete a role', async () => {
+      // roleRepository
+      //   .softDelete('ae032b1b-cc3c-4e44-9197-276ca877a7f8')
+      //   .resolves(Arg.any());
+      groupRoleRepository.createQueryBuilder().returns(groupRoleQueryBuilder);
+      groupRoleQueryBuilder
+        .innerJoinAndSelect(Group, 'group', 'group.id = GroupRole.groupId')
+        .returns(groupRoleQueryBuilder);
+      groupRoleQueryBuilder
+        .where('GroupRole.roleId= :id', {
+          id: 'ae032b1b-cc3c-4e44-9197-276ca877a7f8',
+        })
+        .returns(groupRoleQueryBuilder);
+      groupRoleQueryBuilder.getCount().resolves(0);
+      const resp = await roleService.deleteRole(
+        'ae032b1b-cc3c-4e44-9197-276ca877a7f8',
+      );
+      expect(resp).toEqual(roles[0]);
+    });
   });
 
-  it('should delete a role', async () => {
-    rolesRepository
-      .softDelete('ae032b1b-cc3c-4e44-9197-276ca877a7f8')
-      .resolves(Arg.any());
-    groupRoleRepository.createQueryBuilder().returns(groupRoleQueryBuilder);
-    groupRoleQueryBuilder
-      .innerJoinAndSelect(Group, 'group', 'group.id = GroupRole.groupId')
-      .returns(groupRoleQueryBuilder);
-    groupRoleQueryBuilder
-      .where('GroupRole.roleId= :id', {
-        id: 'ae032b1b-cc3c-4e44-9197-276ca877a7f8',
-      })
-      .returns(groupRoleQueryBuilder);
-    groupRoleQueryBuilder.getCount().resolves(0);
-    const resp = await roleService.deleteRole(
-      'ae032b1b-cc3c-4e44-9197-276ca877a7f8',
-    );
-    expect(resp).toEqual(roles[0]);
+  describe('updateRolePermissions', () => {
```

Comment:
Test fails.

Reply:
Will fix service related tests in a separate PR.